core/src/test/java/fr/sncf/osrd/envelope_sim/AllowanceTests.java

```diff
@@ -224,4 +227,90 @@ public void testConstructionAllowanceOnSegment(double value) {
         assertEquals(speedFirstPointBase, speedFirstPoint, speedFirstPointBase * tolerance);
         assertEquals(speedSecondPointBase, speedSecondPoint, speedSecondPointBase * tolerance);
     }
+
+    /** Tests mareco on a short path where we can't reach max speed,
+     * we only check internal asserts (convergence, envelope asserts) */
+    @Test
+    public void testShortMareco() {
+        var testRollingStock = TestTrains.REALISTIC_FAST_TRAIN;
+        var testPath = new FlatPath(100, 0);
+        makeSimpleMarecoEnvelope(testRollingStock, testPath, 100, new AllowanceValue.Percentage(10), false);
+    }
+
+    /** Test mareco starting in a deceleration section */
+    @Test
+    public void testMarecoStartDeceleration() {
+        var testRollingStock = TestTrains.REALISTIC_FAST_TRAIN;
+        var testPath = new FlatPath(100000, 0);
+        var stops = new double[] { 6000, testPath.getLength() };
+
+        var maxEffortEnvelope = makeComplexMaxEffortEnvelope(testRollingStock, testPath, stops);
+        double start = 0;
+        for (var part : maxEffortEnvelope) {
+            if (part.meta instanceof MaxSpeedEnvelope.DecelerationMeta) {
+                start = (part.getBeginPos() + part.getEndPos()) / 2;
+                break;
+            }
+        }
+        assert start > 0;
+        var allowanceValue = new AllowanceValue.TimePerDistance(10);
+        var allowance = new MarecoAllowance(testRollingStock, testPath, TIME_STEP,
+                start, testPath.getLength(), 0, allowanceValue);
+        allowance.apply(maxEffortEnvelope, stops);
+    }
+
+    /** Test mareco ending in an acceleration section */
+    @Test
+    public void testMarecoEndAcceleration() {
+        var testRollingStock = TestTrains.REALISTIC_FAST_TRAIN;
+        var testPath = new FlatPath(100000, 0);
+        var stops = new double[] { 6000, testPath.getLength() };
+
+        var maxEffortEnvelope = makeComplexMaxEffortEnvelope(testRollingStock, testPath, stops);
+        double end = 0;
+        for (var part : maxEffortEnvelope) {
+            if (part.meta instanceof MaxEffortEnvelope.AccelerationMeta) {
+                end = (part.getBeginPos() + part.getEndPos()) / 2;
+            }
+        }
+        assert end > 0;
+        var allowanceValue = new AllowanceValue.TimePerDistance(10);
+        var allowance = new MarecoAllowance(testRollingStock, testPath, TIME_STEP,
+                0, end, 0, allowanceValue);
+        allowance.apply(maxEffortEnvelope, stops);
+    }
+
+    @Test
+    @Disabled("https://github.com/DGEXSolutions/osrd/issues/565")
```

Comment:
I left a comment on your issue, i think it is a duplicate of an existing one, but i might be wrong here so have a look.

Reply:
Since we have a unit test we can check easily, we can fix the other and see if it fixes this test.